File: backend/schemas/document.py
from pydantic import BaseModel
from datetime import datetime


class DocumentCreateRequest(BaseModel):
    title: str | None = None


class DocumentUpdateRequest(BaseModel):
    title: str | None = None
    content: str | None = None
    content_format: str | None = None
    status: str | None = None


class DocumentOut(BaseModel):
    id: int
    title: str
    content: str | None
    content_format: str
    status: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/document.ts
/**
 * 文档数据结构：与文档列表、编辑器、智能体助手共用。
 * 持久化在 WorkspaceContext 的 localStorage 中。
 */
export interface DocumentItem {
  id: string;
  title: string;
  content: string;
  updatedAt: number;
}
